presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -217,14 +217,12 @@ private QueryContext createQueryContext(
     @Override
     public synchronized void updateMemoryPoolAssignments(MemoryPoolAssignmentsRequest assignments)
     {
-        if (coordinatorId != null && coordinatorId.equals(assignments.getCoordinatorId()) && assignments.getVersion() <= currentMemoryPoolAssignmentVersion) {
+        String assignmentCoordinatorId = assignments.getCoordinatorId();
+        long assignmentVersion = assignments.getVersion();
+        if (assignmentVersion <= coordinatorToPoolAssigmentVersion.getOrDefault(assignmentCoordinatorId, Long.MIN_VALUE)) {
             return;
         }
-        currentMemoryPoolAssignmentVersion = assignments.getVersion();
-        if (coordinatorId != null && !coordinatorId.equals(assignments.getCoordinatorId())) {
```

Comment:
Wondering if this is something even a possibility where query task moved from one coordinator to other?

Reply:
What will happen is, two coordinators will simultaneously send assignments to a worker.  Their assignments are all disjoint, because no two coordinators will have the same set of queries, however the previous logic basically presumed that assignment requests will have the same set of queries.  It's fine if the assignment requests are disjoint--they make no difference to the worker.